lib/Jsrt/ChakraCore.h

```diff
@@ -163,4 +163,346 @@ JsGetModuleHostInfo(
+/// <param name="written">Total number of characters written or to be written</param>
+/// <returns>
+///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+/// </returns>
+CHAKRA_API
+    JsWritePropertyIdUtf8(
+        JsPropertyIdRef propertyId,
+        uint8_t* buffer,
+        size_t bufferSize,
+        size_t* length);
+
+/// <summary>
+///     Serializes a parsed script to a buffer than can be reused.
+/// </summary>
+/// <remarks>
+///     <para>
+///     <c>JsSerializeScript</c> parses a script and then stores the parsed form of the script in a
+///     runtime-independent format. The serialized script then can be deserialized in any
+///     runtime without requiring the script to be re-parsed.
+///     </para>
+///     <para>
+///     Requires an active script context.
+///     </para>
+/// </remarks>
+/// <param name="script">The script to serialize</param>
+/// <param name="buffer">The buffer to put the serialized script into. Can be null.</param>
+/// <param name="bufferSize">
+///     On entry, the size of the buffer, in bytes; on exit, the size of the buffer, in bytes,
+///     required to hold the serialized script.
+/// </param>
+/// <param name="parseAttributes">Encoding for the script.</param>
+/// <returns>
+///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+/// </returns>
+CHAKRA_API
+    JsSerialize(
+        JsValueRef script,
+        BYTE *buffer,
+        unsigned int *bufferSize,
+        JsParseScriptAttributes parseAttributes);
+
+/// <summary>
+///     Parses a serialized script and returns a function representing the script.
+///     Provides the ability to lazy load the script source only if/when it is needed.
+/// </summary>
+/// <remarks>
+///     <para>
+///     Requires an active script context.
+///     </para>
+///     <para>
+///     The runtime will hold on to the buffer until all instances of any functions created from
+///     the buffer are garbage collected.  It will then call scriptUnloadCallback to inform the
+///     caller it is safe to release.
+///     </para>
+/// </remarks>
+/// <param name="buffer">The serialized script.</param>
+/// <param name="scriptLoadCallback">Callback called when the source code of the script needs to be loaded. This is an optional parameter, set to null if not needed.</param>
+/// <param name="sourceContext">
+///     A cookie identifying the script that can be used by debuggable script contexts.
+///     This context will passed into scriptLoadCallback and scriptUnloadCallback.
+/// </param>
+/// <param name="sourceUrl">The location the script came from.</param>
+/// <param name="result">A function representing the script code.</param>
+/// <returns>
+///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+/// </returns>
+CHAKRA_API
+    JsCompileSerialized(
+        BYTE *buffer,
+        JsSerializedLoadScriptCallback scriptLoadCallback,
```

Comment:
Nitpick - put scriptLoadCallback as the first arg similar as [JsParseSerializedScriptWithCallback](https://github.com/Microsoft/ChakraCore/wiki/JsParseSerializedScriptWithCallback)